src/permissions.js

```diff
@@ -5,13 +5,105 @@
  * @module webScience.permissions
  */
 
+/**
+ * An object where keys are Content Security Policy directive names and values are arrays of directive values.
+ * @typedef {Object} ContentSecurityPolicy
+ * @example
+ * {
+ *   "script-src": [ "'self'", "www.example.com"],
+ *   "object-src": [ 'self' ]
+ * }
+ */
+
+/**
+ * The Content Security Policy directives specified in the Content Security Policy Level 3 Working Draft.
+ * @constant {Set<string>}
+ * @private
+ */
+const contentSecurityPolicyDirectives = new Set([ "child-src", "connect-src", "default-src", "font-src", "frame-src", "img-src", "manifest-src", "media-src", "object-src", "prefetch-src", "script-src", "script-src-elem", "script-src-attr", "style-src", "style-src-attr", "worker-src" ]);
```

Comment:
Urgh, I mean it's fine, but it sure would be nice if these were constants we could reference from somewhere instead of having to hardcode a ton of strings (above and below too not just this set).

There is https://www.npmjs.com/package/http-headers-js but I don't know if it's really worth dragging in a third-party dependency.

Reply:
I started by looking for a good quality, actively maintained, and popular library that would handle our requirements... and came up empty. That's why I implemented from scratch. 😕